catroid/src/androidTest/java/org/catrobat/catroid/test/embroidery/DSTFileGeneratorTest.java

```diff
@@ -0,0 +1,214 @@
+		stitchpoints.add(new PointF(-100, 100));
+		stitchpoints.add(new PointF(-100, 0));
+		stitchpoints.add(new PointF(100, 100));
+		stitchpoints.add(new PointF(-100, 100));
+		stitchpoints.add(new PointF(100, 0));
+		if (dstFile.exists()) {
+			dstFile.delete();
+		}
+		dstFile.createNewFile();
+		fileGenerator.writeToDSTFile(dstFile, stitchpoints);
+
+		InputStream inputStream = InstrumentationRegistry.getContext().getResources().openRawResource(org.catrobat
+				.catroid.test.R.raw.sample_dst_file);
+		File compareFile = StorageOperations.copyStreamToDir(inputStream, Constants.CACHE_DIR, "sample_dst_file.dst");
+
+		assertEquals(compareFile.length(), dstFile.length());
+
+		byte[] compareFileBytes = FileUtils.readFile(compareFile);
+		byte[] dstFileBytes = FileUtils.readFile(dstFile);
+
+		boolean fileBytesEquality = Arrays.equals(compareFileBytes, dstFileBytes);
+		assertTrue(fileBytesEquality);
+	}
+
+	@Test
+	public void testPrepareStitchPoints() {
+		final int expectedAmountJumpPoints = 2;
+		EmbroideryList points = new EmbroideryList();
+		points.add(new PointF(0, 0));
+		points.add(new PointF(1000, 1000));
+		ArrayList<DSTFileGenerator.StitchPoint> stitchPoints = fileGenerator.prepareStitchPoints(points);
+
+		assertEquals(points.size() + expectedAmountJumpPoints, stitchPoints.size());
+
+		int index = 0;
+		for (int i = 0; i < stitchPoints.size(); i++) {
+			if (i > 0) {
+				assertTrue(Math.abs(stitchPoints.get(i - 1).point.x - stitchPoints.get(i).point.x) < DSTFileGenerator
+						.MAXDISTANCE);
+				assertTrue(Math.abs(stitchPoints.get(i - 1).point.y - stitchPoints.get(i).point.y) < DSTFileGenerator
+						.MAXDISTANCE);
+			}
+			if (!stitchPoints.get(i).jump) {
+				PointF point = (PointF) points.get(index);
+				assertEquals(stitchPoints.get(i).point.x, point.x * DSTFileGenerator.STITCHPOINTUNITFACTOR);
+				assertEquals(stitchPoints.get(i).point.y, point.y * DSTFileGenerator.STITCHPOINTUNITFACTOR);
+				index += 1;
+			}
+		}
+	}
+
+	@Test
+	public void testHeaderBytes() {
+		final String expectedHeaderString = "LA:testProject    \n" + (char) 0x1A + ("ST:3     \n" + (char) 0x1A
+				+ "CO:1 \n" + (char) 0x1A + "+X:20  \n" + (char) 0x1A + "-X:0   \n" + (char) 0x1A + "+Y:20  \n"
+				+ (char) 0x1A + "-Y:0   \n" + (char) 0x1A + "AX:0    \n" + (char) 0x1A + "AY:0    \n" + (char) 0x1A
+				+ "MX:0    \n" + (char) 0x1A + "MY:0    \n" + (char) 0x1A + "PD:*****\n"
+				+ (char) 0x1A).replace(' ', '\0');
+
+		EmbroideryList points = new EmbroideryList();
+		points.add(new PointF(0, 0));
+		points.add(new PointF(100, 100));
+		points.add(new PointF(0, 0));
+
+		ArrayList<DSTFileGenerator.StitchPoint> stitchPoints = fileGenerator.prepareStitchPoints(points);
+		byte[] header = fileGenerator.getHeaderBytes(stitchPoints);
+		byte[] expectedHeader = expectedHeaderString.getBytes();
+
+		assertEquals(DSTFileGenerator.HEADERMAXBYTE, header.length);
+
+		for (int i = 0; i < DSTFileGenerator.HEADERMAXBYTE; i++) {
```

Comment:
please no loops and branching statements in tests.
you could make the expected header also include the padding, or use substring on header and assert that first substring up to lenght matches expected and the remaining substring only consists of padding.